system/free_tcp_port.go

```diff
@@ -1,18 +1,33 @@
 package system
 
 import (
-	"net"
-	"strconv"
+        "net"
+        "strconv"
+        "errors"
 )
 
-func FindFreePort() (int, error) {
-	l, _ := net.Listen("tcp", ":0")
-	defer l.Close()
+func FindFreePort(num int) (int, error) {
+        t := strconv.Itoa(num)
+        a := ":" + t
+        l, err := net.Listen("tcp",a)
+        if err != nil {
+                return -1, err
+        }
+        parsedPort, _ := strconv.ParseInt(l.Addr().String()[5:], 10, 32)
+        return int(parsedPort), nil
+}
+
 
-	parsedPort, parseErr := strconv.ParseInt(l.Addr().String()[5:], 10, 32)
-	if parseErr != nil {
-		return -1, parseErr
-	}
+func FindFreeInRangePort(Maxport int, Minport int)(int,error) {
+        i := Minport
+        port, err :=FindFreePort(i)
```

Comment:
Please consider returning an error if the user mis configures a max smaller than the min parameter
& add a test case to cover it.